src/target/Purchase/Action.php

```diff
@@ -162,4 +163,25 @@ private function saveInitialUses(SaleInterface $sale, array $initialUses): void
             );
         }
     }
+
+    private function getActiveSales(TargetInterface $target, CustomerInterface $customer, DateTimeImmutable $time = null): ?array
+    {
+        $_sales = $this->saleRepo->findAll((new Specification)->where(array_filter([
+            'seller-id' => $customer->getSeller()->getId(),
+            'target-id' => $target->getId(),
+        ])));
+        if ($time === null || empty($_sales)) {
+            return $_sales;
+        }
+
+        foreach ($_sales as $sale) {
+            if ($sale->getCloseTime() !== null && $sale->getCloseTime() <= $time) {
+                continue;
+            }
+
+            $sales[] = $sale;
+        }
+
+        return $sales;
+    }
```

Comment:
Looks like a copy-paste. Should be moved to a `SaleRepositoryInterface` and implemented in `SaleRepository`.